### ISSUE TITLE
[BUG] exit with code 1

### ISSUE THREAD

**Issue by Xnhyacinth** [microsoft/DeepSpeed#5218]
**Describe the bug**
Running the code [https://github.com/yizhongw/Tk-Instruct](https://github.com/yizhongw/Tk-Instruct) on the 3090. It works fine on the other machine, but on this machine the kill. The training info is as follows:
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   optimizer_legacy_fusion ...... False
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   optimizer_name ............... adamw
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   optimizer_params ............. {'lr': 5e-05, 'betas': [0.9, 0.999], 'eps': 1e-08, 'weight_decay': 0.0}
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   pipeline ..................... {'stages': 'auto', 'partition': 'best', 'seed_layers': False, 'activation_checkpoint_interval': 0}
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   pld_enabled .................. False
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   pld_params ................... False
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   prescale_gradients ........... False
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   scheduler_name ............... WarmupLR
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   scheduler_params ............. {'warmup_min_lr': 0, 'warmup_max_lr': 5e-05, 'warmup_num_steps': 0}
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   sparse_attention ............. None
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   sparse_gradients_enabled ..... False
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   steps_per_print .............. 100000.0
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   train_batch_size ............. 4
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   train_micro_batch_size_per_gpu  1
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   use_node_local_storage ....... False
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   wall_clock_breakdown ......... False
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   world_size ................... 2
[2024-03-01 04:51:13,576] [INFO] [config.py:964:print]   zero_allow_untested_optimizer  False
[2024-03-01 04:51:13,577] [INFO] [config.py:964:print]   zero_config .................. stage=2 contiguous_gradients=True reduce_scatter=True reduce_bucket_size=200000000 allgather_partitions=True allgather_bucket_size=200000000 overlap_comm=True load_from_fp32_weights=True elastic_checkpoint=False offload_param=None offload_optimizer=DeepSpeedZeroOffloadOptimizerConfig(device='cpu', nvme_path=None, buffer_count=4, pin_memory=True, pipeline=False, pipeline_read=False, pipeline_write=False, fast_init=False) sub_group_size=1,000,000,000 cpu_offload_param=None cpu_offload_use_pin_memory=None cpu_offload=None prefetch_bucket_size=50,000,000 param_persistence_threshold=100,000 model_persistence_threshold=sys.maxsize max_live_parameters=1,000,000,000 max_reuse_distance=1,000,000,000 gather_16bit_weights_on_model_save=False stage3_gather_fp16_weights_on_model_save=False ignore_unused_parameters=True legacy_stage1=False round_robin_gradients=False zero_hpz_partition_size=1 zero_quantized_weights=False zero_quantized_gradients=False mics_shard_size=-1 mics_hierarchical_params_gather=False memory_efficient_linear=True
[2024-03-01 04:51:13,577] [INFO] [config.py:964:print]   zero_enabled ................. True
[2024-03-01 04:51:13,577] [INFO] [config.py:964:print]   zero_force_ds_cpu_optimizer .. True
[2024-03-01 04:51:13,577] [INFO] [config.py:964:print]   zero_optimization_stage ...... 2
[2024-03-01 04:51:13,577] [INFO] [config.py:950:print_user_config]   json = {
    "bfloat16": {
        "enabled": "auto"
    }, 
    "fp16": {
        "enabled": false, 
        "loss_scale": 0, 
        "loss_scale_window": 1000, 
        "initial_scale_power": 16, 
        "hysteresis": 2, 
        "min_loss_scale": 1
    }, 
    "optimizer": {
        "type": "AdamW", 
        "params": {
            "lr": 5e-05, 
            "betas": [0.9, 0.999], 
            "eps": 1e-08, 
            "weight_decay": 0.0
        }
    }, 
    "scheduler": {
        "type": "WarmupLR", 
        "params": {
            "warmup_min_lr": 0, 
            "warmup_max_lr": 5e-05, 
            "warmup_num_steps": 0
        }
    }, 
    "zero_optimization": {
        "stage": 2, 
        "offload_optimizer": {
            "device": "cpu", 
            "pin_memory": true
        }, 
        "allgather_partitions": true, 
        "allgather_bucket_size": 2.000000e+08, 
        "overlap_comm": true, 
        "reduce_scatter": true, 
        "reduce_bucket_size": 2.000000e+08, 
        "contiguous_gradients": true
    }, 
    "gradient_accumulation_steps": 2, 
    "gradient_clipping": 1.0, 
    "train_batch_size": 4, 
    "train_micro_batch_size_per_gpu": 1, 
    "steps_per_print": 1.000000e+05
}
[2024-03-01 04:51:17,764] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 1937
[2024-03-01 04:51:17,765] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 1938
[2024-03-01 04:51:18,227] [ERROR] [launch.py:321:sigkill_handler] ['/usr/local/bin/python', '-u', 'src/run_s2s.py', '--local_rank=1', '--do_train', '--do_predict', '--predict_with_generate', '--model_name_or_path', 'google/t5-base-lm-adapt', '--max_source_length', '1024', '--max_target_length', '128', '--generation_max_length', '128', '--max_num_instances_per_task', '100', '--max_num_instances_per_eval_task', '100', '--add_task_name', 'False', '--add_task_definition', 'True', '--num_pos_examples', '2', '--num_neg_examples', '0', '--add_explanation', 'False', '--tk_instruct', 'False', '--data_dir', 'data/splits/default', '--task_dir', 'data/tasks', '--output_dir', 'output/', '--overwrite_output_dir', '--cache_dir', './cache/', '--overwrite_cache', '--per_device_train_batch_size', '1', '--per_device_eval_batch_size', '2', '--gradient_accumulation_steps', '2', '--learning_rate', '5e-05', '--num_train_epochs', '2', '--lr_scheduler_type', 'constant', '--warmup_steps', '0', '--logging_strategy', 'steps', '--logging_steps', '500', '--evaluation_strategy', 'no', '--save_strategy', 'steps', '--save_steps', '2500', '--deepspeed', 'ds_configs/stage2.config', '--bf16', '--run_name', 't5-experiment'] exits with return code = 1

**ds_report output**
[2024-03-01 07:23:01,620] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
--------------------------------------------------
DeepSpeed C++/CUDA extension op report
--------------------------------------------------
NOTE: Ops not installed will be just-in-time (JIT) compiled at
      runtime if needed. Op compatibility means that your system
      meet the required dependencies to JIT install the op.
--------------------------------------------------
JIT compiled ops requires ninja
ninja .................. [92m[OKAY][0m
--------------------------------------------------
op name ................ installed .. compatible
--------------------------------------------------
[93m [WARNING] [0m async_io requires the dev libaio .so object and headers but these were not found.
[93m [WARNING] [0m async_io: please install the libaio-dev package with apt
[93m [WARNING] [0m If libaio is already installed (perhaps from source), try setting the CFLAGS and LDFLAGS environment variables to where it can be found.
async_io ............... [93m[NO][0m ....... [93m[NO][0m
cpu_adagrad ............ [93m[NO][0m ....... [92m[OKAY][0m
cpu_adam ............... [93m[NO][0m ....... [92m[OKAY][0m
fused_adam ............. [93m[NO][0m ....... [92m[OKAY][0m
fused_lamb ............. [93m[NO][0m ....... [92m[OKAY][0m
quantizer .............. [93m[NO][0m ....... [92m[OKAY][0m
random_ltd ............. [93m[NO][0m ....... [92m[OKAY][0m
[93m [WARNING] [0m please install triton==1.0.0 if you want to use sparse attention
sparse_attn ............ [93m[NO][0m ....... [93m[NO][0m

**Comment by Huangzisu** [comment]
Hi! @Xnhyacinth  Could you please tell me how did you fix that? I met the same exit code.
